scripts/tests/test-ci-all.sh

```diff
@@ -74,17 +74,26 @@ function load_test_tool_test() {
 export -f load_test_tool_test
 
 function backend_test_bitcoind() {
-  fm-run-test "${FUNCNAME[0]}" env FM_TEST_ONLY=bitcoind ./scripts/tests/backend-test.sh
+  # backend tests don't support different versions, so we skip for backwards-compatibility tests
```

Comment:
I'm confused why we need to touch anything here.

Reply:
I think you answered your own question in the next comment 😄 
> Ohhhhh... So we're running all tests as a "backward compat test"?

In case it's still unclear, running `test-ci-all` inside the backwards-compatibility tests triggers the backend tests. This PR only adds different versioned binaries, so there isn't a way to use different versioned clients/daemons in our rust code. Running the backend tests will only test the current version, so I'm explicitly skipping since that's already tested in CI.